### PR TITLE
Improve talent portal

### PR DESCRIPTION
**Pull request by milstan**
- add devkit authentication
- limit to PDF files
- add github profile field

### CODE REVIEW

File: common/js/submit.js [hunk 1]
```diff
@@ -38,16 +38,19 @@ if (formElement) {
...
     // register consent
+    const consent_headers = new Headers();
+    consent_headers.append('Authorization','Bearer eyJhbGciOiJFZERTQSIsInR5cCI6InVzZXIifQ.eyJhcHAiOiJjN2M1YTk5OC04YjdiLTQ3YTQtYjc3ZS1jY2ViYTU5NmJlYzQiLCJ1aWQiOiIiLCJleHAiOjE5OTk5OTk5OTl9.7aDXxvP_D3kriH3UxTV_3lxSUT8Za8VSkm4yHP3NoHCUrZ2DEj6gJYUEWJ_hLeojuB6PUaqpGi8XLUSqdv1eDg');
 
     fetch('https://devkit-pce-staging.azurewebsites.net/v0/user-events/consent', {
       method: 'POST',
+      headers: consent_headers,
       body: JSON.stringify({
           dataSubject: {
             id: email,
```
Comment (m4rk055) [inline common/js/submit.js]: This email value will be ignored, as user id is extracted from the token. Token needs to be generated for the specific email or you can generate an app scoped token (dangerous!) and use [this endpoint](https://devkit-pce-staging.azurewebsites.net/swagger/#/User%20events/postV0User-eventsConsentStore).

**Review by m4rk055 — CHANGES_REQUESTED** [review]

File: common/js/submit.js (hunk 1 above)
Comment (milstan) [reply common/js/submit.js]: @m4rk055 changed. See if OK now.

File: common/js/submit.js [hunk 2]
```diff
@@ -1,25 +1,93 @@
...
+}
+
+function redirectError(message){
+  console.log(message);
+  window.location.href = './whoops.html';
+}
 
+function getDevKitToken(email){
+  const builder = window.blindnetTokenGenerator.TokenBuilder.init('c7c5a998-8b7b-47a4-b77e-cceba596bec4', `GfCdj9+mqKdNjhY6sC6H0aAy1JyJu3iQwrc9PdBikg4=`);
+  return tokenPromise = builder.user(email);
```
Comment (m4rk055) [inline common/js/submit.js]:
```suggestion
  const tokenPromise = builder.user(email);
  return tokenPromise;
```

File: common/js/submit.js [hunk 3]
```diff
@@ -32,76 +100,35 @@ if (formElement) {
...
-    .catch(error => {
-      console.error('Error:', error);
+
+    //generate devkit token
+
+      getDevKitToken(email).then(token => {
+      console.log('token '+token);
+
+      //if OK register consent
+      registerConsent(token).then((response) => {
```
Comment (m4rk055) [inline common/js/submit.js]:
```suggestion
      return registerConsent(token).then((response) => {
```

File: common/js/submit.js [hunk 4]
```diff
@@ -1,25 +1,93 @@
...
+  headers.append('Content-Type','application/json');
+  headers.append('Prefer','return=representation');
+
+  var object = {};
+  formData.forEach((value, key) => object[key] = value);
+  object['cv'] = reader.result;
+  console.log(object['cv']);
+  var body = JSON.stringify(object);
+
+  fetch('https://jirxtzamlvwamuyigzmj.supabase.co/rest/v1/cv-sumbissions', {
```
Comment (m4rk055) [inline common/js/submit.js]:
```suggestion
  return fetch('https://jirxtzamlvwamuyigzmj.supabase.co/rest/v1/cv-sumbissions', {
```

File: common/js/submit.js [hunk 5]
```diff
@@ -32,76 +100,35 @@ if (formElement) {
...
+
+      getDevKitToken(email).then(token => {
+      console.log('token '+token);
+
+      //if OK register consent
+      registerConsent(token).then((response) => {
+        if (response.ok) {
+          console.log(`consent registered`);
+          //if OK submit data
+          submitData(formData);
```
Comment (m4rk055) [inline common/js/submit.js]:
```suggestion
          return submitData(formData);
```

**Comment by milstan** [comment]
changes you requested are in last commit @m4rk055 feelf free to merge if happy with them

**Comment by milstan** [comment]
@m4rk055 there is an issue - the token is generated, but the consent endpoint returns 401 unauthroised

**Review by m4rk055 — APPROVED** [review]

**Comment by m4rk055** [comment]
> @m4rk055 there is an issue - the token is generated, but the consent endpoint returns 401 unauthroised

Fixed

**Pull request merged**